lib/cc/analyzer/container.rb

```diff
@@ -0,0 +1,98 @@
+require "posix/spawn"
+
+module CC
+  module Analyzer
+    class Container
+      DEFAULT_TIMEOUT = 15 * 60 # 15m
+
+      def initialize(
+        image:,
+        name:,
+        command: nil,
+        log: NullContainerLog.new,
+        timeout: DEFAULT_TIMEOUT
+      )
+        @image = image
```

Comment:
I don't have a strong preference between making (private) `attr_reader`s or using instance variables directly. I think I advocated the latter at one point but because I see the former more prevalent our code bases I've been starting to do that -- which is what I've done here.